app/controllers/api/v1/authentication_controller.rb

```diff
@@ -55,24 +55,23 @@ class AuthenticationController < Api::V1::ApiController
 
     ===Response status codes
     * 200 - returned regardless of whether mobile number exists or not
+    * 403 - returned if the host is invalid
     * 422 - returned if the mobile number is invalid
     EOS
     param :mobile, String, desc: "Mobile number with prefixed country code e.g. +85262345678"
     error 422, "Invalid mobile number - if mobile prefix doesn't start with +852"
+    error 403, "Invalid host"
     error 500, "Internal Server Error"
     # Lookup user based on mobile. Validate mobile format first.
     def send_pin
-      mobile = params[:mobile]
-      if (User::HongKongMobileRegExp === mobile)
-        @user = User.find_by_mobile(params[:mobile])
-        @user.send_verification_pin if @user.present?
+      @mobile = params[:mobile]
+      if(User::HongKongMobileRegExp === @mobile)
+        @user = User.find_by_mobile(@mobile)
+        return render_error(I18n.t('host.invalid'), 403) if !valid_host?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.